pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/ReaderThread.java

```diff
@@ -146,13 +146,14 @@ protected void skipFirstMessageIfNeeded() throws PulsarClientException {
             if (!messageIdRoughEquals(startMessageId, lastMessageId) && !reader.hasMessageAvailable()) {
                 MessageIdImpl startMsgIdImpl = (MessageIdImpl) startMessageId;
                 // startMessageId is bigger than lastMessageId
-                if (!metaDataReader.checkCursorAvailable(reader.getTopic(), startMsgIdImpl)) {
+                if (startMsgIdImpl.compareTo(lastMessageId) > 0) {
                     if (failOnDataLoss) {
                         log.error("the start message id is beyond the last commit message id, with topic:{}", this.topicRange);
                         throw new RuntimeException("start message id beyond the last commit");
                     } else {
-                        log.info("reset message to valid offset {}", startMessageId);
-                        metaDataReader.resetCursor(this.topicRange, startMessageId);
+                        log.info("reset message to valid offset {}", lastMessageId);
```

Comment:
Just a note here, user may want to reset it to the earliest available message.

We may implement later if it is requested.

Reply:
> Just a note here, user may want to reset it to the earliest available message.
> 
> We may implement later if it is requested.

@nlu90 For the case of startMsgIdImpl> lastMessageId, lastMessageId will now be used to read the message. You can add options to allow users to reset to the earliest position.